osu.Game/Rulesets/Mods/Mod.cs

```diff
@@ -148,15 +150,96 @@ public void CopyFrom(Mod source)
+                var targetBindable = (IBindable)property.GetValue(this)!;
+                var sourceBindable = (IBindable)property.GetValue(source)!;
 
                 CopyAdjustedSetting(targetBindable, sourceBindable);
             }
         }
 
+        /// <summary>
+        /// Copies all mod setting values sharing same <see cref="MemberInfo.Name"/> from <paramref name="source"/> into this instance.
+        /// </summary>
+        /// <param name="source">The mod to copy properties from.</param>
+        internal void CopySharedSettings(Mod source)
+        {
+            const string min_value = nameof(BindableNumber<int>.MinValue);
+            const string max_value = nameof(BindableNumber<int>.MaxValue);
+            const string value = nameof(Bindable<int>.Value);
+
+            Dictionary<string, object> oldSettings = new Dictionary<string, object>();
+
+            foreach (var (_, property) in source.GetSettingsSourceProperties())
+            {
+                oldSettings.Add(property.Name.ToSnakeCase(), property.GetValue(source)!);
+            }
+
+            foreach (var (_, property) in this.GetSettingsSourceProperties())
+            {
+                object targetSetting = property.GetValue(this)!;
+
+                if (!oldSettings.TryGetValue(property.Name.ToSnakeCase(), out object? sourceSetting))
+                    continue;
+
+                if (((IBindable)sourceSetting).IsDefault)
+                    // keep at default value if the source is default
+                    continue;
+
+                Type? targetType = getGenericBaseType(targetSetting, typeof(BindableNumber<>));
+                Type? sourceType = getGenericBaseType(sourceSetting, typeof(BindableNumber<>));
+
+                if (targetType == null || sourceType == null)
+                {
+                    if (getGenericBaseType(targetSetting, typeof(Bindable<>))!.GenericTypeArguments.Single() ==
+                        getGenericBaseType(sourceSetting, typeof(Bindable<>))!.GenericTypeArguments.Single())
+                    {
+                        // change settings only if the type is the same
+                        setValue(targetSetting, value, getValue(sourceSetting, value));
+                    }
+
+                    continue;
+                }
+
+                Type targetGenericType = targetType.GenericTypeArguments.Single();
+                Type sourceGenericType = sourceType.GenericTypeArguments.Single();
+
+                double allowedMin = Math.Max(
+                    Convert.ToDouble(targetGenericType.GetField("MinValue")!.GetValue(null)),
+                    Convert.ToDouble(sourceGenericType.GetField("MinValue")!.GetValue(null))
+                );
+
+                double allowedMax = Math.Min(
+                    Convert.ToDouble(targetGenericType.GetField("MaxValue")!.GetValue(null)),
+                    Convert.ToDouble(sourceGenericType.GetField("MaxValue")!.GetValue(null))
+                );
```

Comment:
`RangeConstrainedBindable.HasDefinedRange` is public. You can just use that and check if the target bindable has a range to rescale to. If not, then you can transfer values raw from the source.

Reply:
Alright, I didn't know about that. Definitely sound like a better solution!

Reply:
Thinking about it now, I have a feeling I can't get rid of the `allowedMin`/`allowedMax` values. Take for example conversion with `sourceValue (double) = int.MaxValue + 1d` trying to convert to `int`, which should either throw an error or clamp the value to `allowedMax` (in this case `int.MaxValue`).

Reply:
I don't know. I'm not convinced by any of this, as I'm not even really sure that any of this rescaling logic even really makes sense. I think I'd rather just refuse to transfer values across if the types and/or bounds don't match and save all of this hemming and hawing over hypothetical situations and weird edge cases.

Reply:
I don't know either. Maybe make someone else have a look at this as well, otherwise I'm fine with just copying the `Value` only if that is the only thing different from the two `Bindable`s.

Reply:
If clamping to the new range is simple then go with that. I think it's better than not copying at all. Probably.

Reply:
That would work and make the code simpler. But what about the scaling of values (i.e. having different ranges between settings)?